src/Aspire.Hosting.Azure/Extensions/AzureEventHubsExtensions.cs

```diff
@@ -0,0 +1,101 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Runtime.Versioning;
+using Aspire.Hosting.ApplicationModel;
+using Aspire.Hosting.Azure;
+using Azure.Provisioning;
+using Azure.Provisioning.Authorization;
+using Azure.Provisioning.EventHubs;
+
+namespace Aspire.Hosting;
+
+/// <summary>
+/// Provides extension methods for adding the Azure Service Bus resources to the application model.
+/// </summary>
+public static class AzureEventHubsExtensions
+{
+    /// <summary>
+    /// Adds an Azure Event Hubs Namespace resource to the application model. This resource can be used to create hub resources.
+    /// </summary>
+    /// <param name="builder">The builder for the distributed application.</param>
+    /// <param name="name">The name of the resource.</param>
+    /// <returns></returns>
+    public static IResourceBuilder<AzureEventHubsResource> AddAzureEventHubs(
+        this IDistributedApplicationBuilder builder, string name)
+    {
+#pragma warning disable CA2252 // This API requires opting into preview features
+        return builder.AddAzureEventHubs(name, null);
+#pragma warning restore CA2252 // This API requires opting into preview features
+    }
+
+    /// <summary>
+    /// Adds an Azure Event Hubs Namespace resource to the application model. This resource can be used to create hub resources.
+    /// </summary>
+    /// <param name="builder">The builder for the distributed application.</param>
+    /// <param name="name">The name of the resource.</param>
+    /// <param name="configureResource">Optional callback to configure the Event Hubs namespace.</param>
+    /// <returns></returns>
+    [RequiresPreviewFeatures]
+    public static IResourceBuilder<AzureEventHubsResource> AddAzureEventHubs(this IDistributedApplicationBuilder builder, string name,
+        Action<IResourceBuilder<AzureEventHubsResource>, ResourceModuleConstruct, EventHubsNamespace>? configureResource)
+    {
+        var configureConstruct = (ResourceModuleConstruct construct) =>
+        {
+            var serviceBusNamespace = new EventHubsNamespace(construct, name: name);
```

Comment:
nit:  The variable names should reflect Event Hubs, not Service Bus.

Reply:
I go back and forth about this since all event hubs namespaces look like *.servicebus.windows.net -- even the Bicep templates expose it as a serviceBusNamespace, right?

Reply:
but I'll change it to be consistent with the dotnet type.

Reply:
While they share a common lineage, Event Hubs namespaces and Service Bus namespaces are two different Azure resources and should be treated  interpedently.   I can't speak to what the Aspire bicep files are doing, but it would be based on a `listKeys` call to the Event Hubs resource.